src/lib/components/market/item_row_wrapper.ts

```diff
@@ -140,11 +141,17 @@ export class ItemRowWrapper extends FloatElement {
             return nothing;
         }
 
+        const fadePercentage = this.asset && this.itemInfo && getFadePercentage(this.asset, this.itemInfo);
```

Comment:
nit: You can move this `fadePercentage` defn to within the `if-block` below which prevents the additional `this.iteminfo` presence check.